zebrad/src/components/sync.rs

```diff
@@ -283,24 +232,12 @@ where
                 // Once we're below the lookahead limit, we can keep extending the tips.
                 tracing::info!(
                     tips.len = self.prospective_tips.len(),
-                    pending.len = self.pending_blocks.len(),
-                    pending.limit = LOOKAHEAD_LIMIT,
+                    in_flight = self.downloads.in_flight(),
+                    lookahead_limit = LOOKAHEAD_LIMIT,
                     "extending tips",
                 );
-                let old_tips = self.prospective_tips.clone();
-                let _ = self.extend_tips().await;
 
-                // If ExtendTips fails, wait, then give it another shot.
-                //
-                // If we don't have many peers, waiting and retrying helps us
-                // ignore unsolicited BlockHashes from peers.
-                if self.prospective_tips.is_empty() {
-                    self.update_metrics();
-                    tracing::info!("no new tips, waiting to retry extend tips");
-                    delay_for(TIPS_RETRY_TIMEOUT).await;
-                    self.prospective_tips = old_tips;
-                    let _ = self.extend_tips().await;
-                }
+                let _ = self.extend_tips().await;
```

Comment:
How is this error handled or mitigated by the changes in this PR?

As far as I understand, the cancellation in `Downloads` mitigates "duplicate block" errors. But I can't see any new code in this PR that handles or mitigates ExtendTips errors.

So I wonder if we'll still need this retry, particularly on testnet.

Reply:
If we fail to extend the tip set, then we run out of prospective tips and restart the sync procedure.  This PR tries to make restarts cheap, so that restarting doesn't cause a cascading failure.

Reply:
Unresolved since I'm not sure this approach actually works on testnet.

Reply:
Yeah I wonder if we should just deploy a few Zebra or zcashd nodes to stabilise testnet.

Or if we should spend the effort to make sync work under difficult network conditions.

Reply:
Can we split the ExtendTips change into another PR, and merge the rest of this PR?

Reply:
@hdevalence can we split the ExtendTips change into another PR?

Reply:
No, I'd rather fix it here.